### PR TITLE
Menu Subsystem Implementation

### PR DESCRIPTION
**Pull request by fr500**
## Description

Allows loading subsystem games from the menu

It highlights the subsystem we're loading to with a star:

image

It tells you which rom for such subsystem is currently being loaded:

image

And once ready the entry changes to Start instead of Load

image

If you pick another subsystem it cleans up the currently loaded roms, and once loaded it cleans up too so there are no regressions that I could find in other loading cases so far

Results:

image

image

## Comments

- You NEED TO HAVE CONTENT LOADED WITH THE CORE for it to be available.
- SUBSYSTEM is set at the same time as **supports no content** so it should be possible to make this work with **Load Core** but I couldn't figure that out
- Some menu strings are non localizable, I don't like doing localization...
- Only available in XMB, I guess adding it to the other menus is just a copy paste exercise
- Doesn't support loading from archives (because I didn't understand how to)
- Maybe it should be in a submenu instead of bloating the main menu
- It's missing left / right / start / select callbacks

## Related Issues

[Any issues this pull request may be addressing]

## Related Pull Requests

[Any other PRs from related repositories that might be needed for this pull request to work]

## Reviewers
@twinaphex

### CODE REVIEW

**Comment by Tatsuya79** [comment]
Perhaps you could use retropad_start/space key when on "Start Super Gameboy" to unload the roms and go back to "Load Super Gameboy".

**Comment by fr500** [comment]
ah sure, just add `content_clear_subsystem` to the start callback, I already spent waaaaay more time that I expected on this and I'm out for the day :P

**Comment by twinaphex** [comment]
We will continue working on this and improving on it on our master branch.

File: intl/msg_hash_lbl.h [hunk 1]
```diff
@@ -361,6 +361,10 @@ MSG_HASH(MENU_ENUM_LABEL_DISK_CYCLE_TRAY_STATUS,
       "disk_cycle_tray_status")
 MSG_HASH(MENU_ENUM_LABEL_DISK_IMAGE_APPEND,
       "disk_image_append")
+MSG_HASH(MENU_ENUM_LABEL_SUBSYSTEM_ADD,
+      "subsystem_add")
```
Comment (fr500) [inline intl/msg_hash_lbl.h]: this is the action that  is triggered when a rom is selected

File: intl/msg_hash_lbl.h [hunk 2]
```diff
@@ -361,6 +361,10 @@ MSG_HASH(MENU_ENUM_LABEL_DISK_CYCLE_TRAY_STATUS,
...
+MSG_HASH(MENU_ENUM_LABEL_SUBSYSTEM_LOAD,
+      "subsystem_load")
```
Comment (fr500) [inline intl/msg_hash_lbl.h]: this is the action that  is triggered when the start subsystem option is selected

File: menu/drivers/xmb.c [hunk 3]
```diff
@@ -4435,6 +4438,46 @@ static int xmb_list_push(void *data, void *userdata,
...
+               {
+                  for (i = 0; i < system->subsystem.size; i++, subsystem++)
+                  {
+                     char s[PATH_MAX_LENGTH];
+                     if (content_get_subsystem() == i)
+                     {
+                        if (content_get_subsystem_rom_id() < subsystem->num_roms)
+                        {
+                           snprintf(s, sizeof(s), "Load %s %s", subsystem->desc, i == content_get_subsystem() ? "\u2605" : " ");
+                           menu_entries_append_enum(info->list,
```
Comment (fr500) [inline menu/drivers/xmb.c]: this adds the Load Subsystem entry to the main menu for the **current active subsystem**
there is a bit of code duplication between this and the next one I usually write code like this, sorry.

File: menu/drivers/xmb.c [hunk 4]
```diff
@@ -4435,6 +4438,46 @@ static int xmb_list_push(void *data, void *userdata,
...
+                                 s,
+                                 msg_hash_to_str(MENU_ENUM_LABEL_SUBSYSTEM_LOAD),
+                                 MENU_ENUM_LABEL_SUBSYSTEM_LOAD,
+                                 MENU_SETTINGS_SUBSYSTEM_LOAD, 0, 0);
+                        }
+                     }
+                     else
+                     {
+                        snprintf(s, sizeof(s), "Load %s %s", subsystem->desc, i == content_get_subsystem() ? "\u2605" : " ");
+                        menu_entries_append_enum(info->list,
```
Comment (fr500) [inline menu/drivers/xmb.c]: this adds the Load Subsystem entry to the main menu for the any not subsystem that is not currently active

Actually I just realized this code will do nothing here:
**i == content_get_subsystem() ? "\u2605" : " "**

**Pull request merged**